pinot-core/src/main/java/org/apache/pinot/core/data/table/TableResizer.java

```diff
@@ -112,12 +112,12 @@
    * For aggregation values in the order by, the final result is extracted if the intermediate result is non-comparable
    */
   @VisibleForTesting
-  IntermediateRecord getIntermediateRecord(Record record) {
+  IntermediateRecord getIntermediateRecord(Key key, Record record) {
```

Comment:
why do we need this method who will invoke it.

Reply:
while doing the resize, the Record gets converted to IntermediateRecord. IntermediateRecord contains only the columns necessary for order by. It helps prevent array index lookup during order by.
Now that we have gotten rid of Key from Record, we might be able to do this using just the Record. Can revisit it after this